test/parallel_api/numeric/numeric.ops/reduce_by_segment.pass.cpp

```diff
@@ -21,210 +21,241 @@
+                              void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last, Iterator2 vals_first, Iterator2 vals_last,
+               Iterator3 key_res_first, Iterator3 key_res_last, Iterator4 val_res_first, Iterator4 val_res_last, Size n)
+    {
+    }
+};
+
+template <typename Key, typename Value, typename TestName>
+void
+test_on_host()
+{
+#if !TEST_DPCPP_BACKEND_PRESENT
+    const int max_n = 100000;
+    const int inout1_offset = 3;
+    const int inout2_offset = 5;
+    const int inout3_offset = 7;
+    const int inout4_offset = 9;
 #endif
+    for (size_t n = 1; n <= max_n; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+    {
+    // create buffers
+    std::vector<Key>   inout1(max_n + inout1_offset);
+    std::vector<Value> inout2(max_n + inout2_offset);
+    std::vector<Key>   inout3(max_n + inout3_offset);
+    std::vector<Value> inout4(max_n + inout4_offset);
+
+    // create iterators
+    auto inout1_offset_first = std::begin(inout1) + inout1_offset;
+    auto inout2_offset_first = std::begin(inout2) + inout2_offset;
+    auto inout3_offset_first = std::begin(inout3) + inout3_offset;
+    auto inout4_offset_first = std::begin(inout4) + inout4_offset;
 
-void test_on_host() {
-    const int N = 7;
-    int A[N] = {1, 3, 3, 3, 2, 2, 1}; // input keys
-    int B[N] = {9, 8, 7, 6, 5, 4, 3}; // input values
-    int C[N];                         // output keys
-    int D[N];                         // output values
-
-    std::pair<int*,int*> new_end;
-    new_end = oneapi::dpl::reduce_by_segment(oneapi::dpl::execution::par, A, A + N, B, C, D, std::equal_to<int>(),
-                                    std::plus<int>());
-    // The first four keys in C are now {1, 3, 2, 1} and new_end.first - C is 4.
-    // The first four values in D are now {9, 21, 9, 3} and new_end.second - D is 4.
-    ASSERT_EQUAL(C[0], 1);
-    ASSERT_EQUAL(C[1], 3);
-    ASSERT_EQUAL(C[2], 2);
-    ASSERT_EQUAL(C[3], 1);
-    ASSERT_EQUAL(D[0], 9);
-    ASSERT_EQUAL(D[1], 21);
-    ASSERT_EQUAL(D[2], 9);
-    ASSERT_EQUAL(D[3], 3);
-    ASSERT_EQUAL(std::distance(C, new_end.first), 4);
-    ASSERT_EQUAL(std::distance(D, new_end.second), 4);
+#if _ONEDPL_DEBUG_SYCL
+        ::std::cout << "n = " << n << ::std::endl;
+#endif
+        invoke_on_all_host_policies()(
+            TestName(), inout1_offset_first, inout1_offset_first + n, inout2_offset_first, inout2_offset_first + n,
+            inout3_offset_first, inout3_offset_first + n, inout4_offset_first, inout4_offset_first + n, n);
+    }
 }
 
-int main() {
+int
+main()
+{
 #if TEST_DPCPP_BACKEND_PRESENT
-    test_with_buffers();
-    test_with_usm();
+    test4buffers<char, uint64_t, test_reduce_by_segment>();
 #endif
-    test_on_host();
-
-    return TestUtils::done();
+    test_on_host<uint64_t, int, test_reduce_by_segment>();
+    return done(TEST_DPCPP_BACKEND_PRESENT);
```

Comment:
The test will provide skipped status with only host testing. Let's change to done()